Dialog/TilesetEditDialog.cpp

```diff
@@ -993,3 +932,272 @@ void TilesetEditDialog::on_pushButton_ImportPalette_clicked()
     ReRenderTile8x8Map(SelectedPaletteId);
     ReRenderTile16Map();
 }
+
+/// <summary>
+/// Inport Tile16 graphic data from bin files.
+/// </summary>
+void TilesetEditDialog::on_pushButton_ImportTile16Graphic_clicked()
+{
+    /** rule(s):
+     *  don't overwrite the first Tile16
+     *  don't let imported Tile16 id goes out of bound (should be smaller than 0x2FF)
+     *  TODO: support flexible Tile16 set length in the future, and we can read each of its length from the RATS struct
+    **/
+    LevelComponents::Tileset *tmp_newTilesetPtr = tilesetEditParams->newTileset;
+    int selTile16 = SelectedTile16;
+    int selPalId = SelectedPaletteId;
+    FileIOUtils::ImportTile8x8GfxData(this,
+        tmp_newTilesetPtr->GetPalettes()[SelectedPaletteId],
+        [selTile16, tmp_newTilesetPtr, selPalId] (QByteArray finaldata, QWidget *parentPtr)
```

Comment:
Try to consolidate this code in FileIOUtils from the multiple callbacks that use it.

Reply:
so the first one find if we can import Tile16 into the Tileset by comparing each existing Tile8x8, and see if the elements in Tile16 match one of them
the second one eliminate replicated Tile8x8 and managing the Tile16 all using the Tile8x8 id in the new arranged Tile8x8 set (edited)
they just looks similar, but not function similarly
(record from private talk)